### ISSUE TITLE
Genie Authentication Issue

### ISSUE THREAD

**Issue by ai-ml-with-kapil** [GenieFramework/Stipple.jl#188]
While using @page (page macro from stipple) It overrides the genie authentication rules and allows a person directly to access routes even which I have bound with authentication.
Sharing sample code to have an idea

in routes.jl ->
route("/") do
  authenticated!()
  App.DemoController.index()
end

in DemoControllers.jl ->

function index()
    if authenticated!()
      @page("/", "app/resources/demo/views/dmeo_ui.jl.html").route.action()
    else
      redirect("/login")
    end
end

only when the server starts for the first time, then only it is bound coz by that time page is not up with @page macro, once it is up, now even if I logout, I can directly use the URL to enter the page and this time it is not redirecting me to login page. Can somebody help?

**Comment by essenciary** [comment]
@ai-ml-with-kapil Thanks for reporting this - we haven't integrated the two yet so they're not known to work, but definitely we need to come up with a solution for this. I'll take a look.

**Comment by essenciary** [comment]
OK, so what I've done in an app was this: 

1/ have a `routes.jl` file which has something like: 
```julia
route("/dashboard", DashboardController.index)
```

2/ In `DashboardController.jl`
```julia
function index()
  authenticated!()

  @page("/dashboard", "app/resources/dashboard/views/index.jl").route.action()
end
```

**Comment by essenciary** [comment]
Let me retest, maybe changes in `@page` broke the flow.

**Comment by hhaensel** [comment]
I think that's the caching behaviour, which I also found to break revising  code with Revise.jl

**Comment by essenciary** [comment]
Can you please give more details @hhaensel ?

**Comment by svilupp** [comment]
> I think that's the caching behaviour, which I also found to break revising code with Revise.jl

Oh, I saw that in your reply in another Issue -- let me try it out!

**Comment by svilupp** [comment]
@hhaensel I don't think it's related to Revise. I've disabled it and the problem persists, but I do agree that it's some caching going wrong - my investigation points to routing tables being changed by `@page`?

I've tried adding auth to the Stipple Iris demo. Here are a few additional observations:
- First time I open it, it requires login; any subsequent reload is logged in. If I open it from incognito mode or a different browser, it's still logged in. I can even shut down the server, start it again and I'm logged in (same session().id though, so perhaps it's saved in cookies? I haven't checked)
- I have added logging to route "/" and I can see the logs only on first execution
- I checked `Router.routes()` and it has the correct function until I try to access "/", after which is changes to a new anonymous function for `:get`
- After which, I can literary access the dashboard from Julia REPL `Router.get_route(:get).action()` (shouldn't be possible as I don't have a logged in session when running from REPL)
- I guess that `@page` overwrites the routing table and skips all auth steps


Snippet I use to add a login to Iris demo (I've disabled the @page macro because I wanted to have more control). I've added some logging to know what's happening.
```
const model = @init

route("/") do
    @info "I'm here and my auth status is: $(is_authenticated()) (sessionid: $(session().id))"
    authenticated!()
    @page("/", "ui.jl").route.action()
end

Server.isrunning() || Server.up()
```
